lib/Controller/DocumentController.php

```diff
@@ -1131,6 +1142,16 @@ private function putRelative($fileId, $owner, $editor, $suggested) {
 			$root->newFolder(\dirname($path));
 		}
 
+		try {
+			$view = new View();
+			$view->verifyPath($path, $suggested);
+		} catch (\Exception $e) {
```

Comment:
Better catch only the `InvalidPathException`, which is the one we're interested in. Other exceptions might require different handling